client/client.go

```diff
@@ -574,6 +574,56 @@ func (r *NotaryRepository) GetDelegationRoles() ([]*data.Role, error) {
 	return allDelegations, nil
 }
 
+// RoleWithSignatures is a Role with its associated signatures
+type RoleWithSignatures struct {
+	Signatures []data.Signature
+	data.Role
+}
+
+// GetRepoRoleMetaInfo returns a list of RoleWithSignatures objects for this repo
+// This represents the latest metadata for each role in this repo
+func (r *NotaryRepository) GetRepoRoleMetaInfo() ([]RoleWithSignatures, error) {
```

Comment:
I was thinking that we would leave this as public functions on the `NotaryRepository`, but by "API" do you mean a server endpoint or notary CLI command?

I was originally thinking that we would have this function for retrieving the status of roles for a repo, and make use of `ListTargets` just for considering the status of targets (the two could be called together in another function as well), but I could also modify this to be a full `GetRepoInfo` as you're suggesting such that we would also only require to call `Update()` once.